Tests/NIOHTTP1Tests/HTTPUpgradeTests.swift

```diff
@@ -755,4 +763,52 @@ class HTTPUpgradeTestCase: XCTestCase {
         let resultString = data.map { $0.getString(at: $0.readerIndex, length: $0.readableBytes)! }.joined(separator: "")
         XCTAssertEqual(resultString, appData)
     }
+
+    func testRemovesOtherHandlersFromPipeline() throws {
+        let g = DispatchGroup()
+        g.enter()
+        var serverChannel: Channel? = nil
+
+        let otherHandler = ExplodingHandler()
+        let upgrader = UpgradeDelayer(forProtocol: "myproto")
+        let handler = HTTPServerUpgradeHandler(upgraders: [upgrader],
+                                               httpEncoder: nil,
+                                               httpDecoder: nil,
+                                               extraHTTPHandlers: [otherHandler]) { ctx in
+                                                serverChannel = ctx.channel
```

Comment:
this is technically not correctly synchronised as you write & read to/from `serverChannel` from different threads without synchronisation. (`DispatchGroup` doesn't technically do any memory synchronisation)